core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -909,32 +922,49 @@ public boolean load() {
+        if (versionStr.isEmpty() || !String.valueOf(Constants.VERSION_EM).equals(versionStr))
+            throw new IllegalStateException("Incompatible encoding version. You need to use the same GraphHopper version you used to import the graph, or run a new import. "
+                    + " Stored encoding version: " + (versionStr.isEmpty() ? "missing" : versionStr) + ", used encoding version: " + Constants.VERSION_EM);
+        String encodedValueStr = properties.get("graph.encoded_values");
+        ArrayNode evList = deserializeEncodedValueList(encodedValueStr);
+        LinkedHashMap<String, EncodedValue> encodedValues = new LinkedHashMap<>();
+        evList.forEach(serializedEV -> {
+            EncodedValue encodedValue = EncodedValueSerializer.deserializeEncodedValue(serializedEV.textValue());
+            if (encodedValues.put(encodedValue.getName(), encodedValue) != null)
+                throw new IllegalStateException("Duplicate encoded value name: " + encodedValue.getName() + " in: graph.encoded_values=" + encodedValueStr);
+        });
 
-            String encodedValueStr = properties.get("graph.encoded_values");
-            if (!encodingManager.toEncodedValuesAsString().equalsIgnoreCase(encodedValueStr)) {
-                throw new IllegalStateException("Encoded values do not match:"
-                        + "\nGraphhopper config: " + encodingManager.toEncodedValuesAsString()
-                        + "\nLoaded graph:       " + encodedValueStr
-                        + "\nChange configuration to match the graph or delete " + baseGraph.getDirectory().getLocation());
-            }
-            baseGraph.loadExisting();
-            return true;
+        String flagEncodersStr = properties.get("graph.flag_encoders");
+        LinkedHashMap<String, VehicleEncodedValues> flagEncoders = Stream.of(flagEncodersStr.split(","))
+                .map(str -> flagEncoderFactory.deserializeFlagEncoder(str, name -> {
+                    EncodedValue ev = encodedValues.get(name);
+                    if (ev == null)
+                        throw new IllegalStateException("FlagEncoder " + str + " uses unknown encoded value: " + name);
+                    return ev;
+                }))
+                .collect(Collectors.toMap(FlagEncoder::getName, f -> (VehicleEncodedValues) f,
+                        (f1, f2) -> {
+                            throw new IllegalStateException("Duplicate flag encoder: " + f1.getName() + " in: " + flagEncodersStr);
+                        },
+                        LinkedHashMap::new));
+
+        EncodedValue.InitializerConfig edgeConfig = EncodedValueSerializer.deserializeInitializerConfig(properties.get("graph.em.edge_config"));
+        EncodedValue.InitializerConfig turnCostConfig = EncodedValueSerializer.deserializeInitializerConfig(properties.get("graph.em.turn_cost_config"));
+        encodingManager = new EncodingManager(encodedValues, flagEncoders, edgeConfig, turnCostConfig);
```

Comment:
I changed this now: We just store the init configs in the properties file and do not use the EM Builder at all anymore when deserializing the EM.